lib/gcov/include/uk/gcov.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#ifndef __UK_GCOV_H__
+#define __UK_GCOV_H__
+
+/* The linker script supplies the start and end symbols.
+ * We employ array syntax to circumvent potential
+ * small-data area problems
+ */
+extern const struct gcov_info *const __gcov_info_start[];
```

Comment:
Actually if these are not used by the caller, they don't need to be part of the public API. If that's the case, you can move them back to `ukgcov.c`. IMO this comment is not necessary, as we declare these using the array notation in several other libraries.